Source/Basic-Conditions-And-Behaviors/Editor/UI/Drawers/Data Property Drawers/ProcessVariableDrawer.cs

```diff
@@ -91,53 +76,81 @@ public override Rect Draw(Rect rect, object currentValue, Action<object> changeV
 
-            if (oldUniqueName != newUniqueName)
+            if (oldGuid != newGuid)
             {
-                RevertableChangesHandler.Do(
-                    new ProcessCommand(
-                        () =>
-                        {
-                            processVariable.PropertyReference.UniqueName = newUniqueName;
-                            changeValueCallback(processVariable);
-                        },
-                        () =>
-                        {
-                            processVariable.PropertyReference.UniqueName = oldUniqueName;
-                            changeValueCallback(processVariable);
-                        }),
-                    isUndoOperation ? undoGroupName : string.Empty);
-
-                if (isUndoOperation)
-                {
-                    RevertableChangesHandler.CollapseUndoOperations(undoGroupName);
-                }
+                processVariable.Property.ResetGuids(new List<Guid>() { newGuid });
+                changeValueCallback(processVariable);
             }
 
             if (newConstValue != null && newConstValue.Equals(processVariable.ConstValue) == false)
             {
-                RevertableChangesHandler.Do(
-                    new ProcessCommand(
-                        () =>
-                        {
-                            processVariable.ConstValue = newConstValue;
-                            changeValueCallback(processVariable);
-                        },
-                        () =>
-                        {
-                            processVariable.ConstValue = oldConstValue;
-                            changeValueCallback(processVariable);
-                        }),
-                    isUndoOperation ? undoGroupName : string.Empty);
-
-                if (isUndoOperation)
+                processVariable.ConstValue = newConstValue;
+                changeValueCallback(processVariable);
+            }
+
+            return rect;
+        }
+
+        private Guid GetIDFromSelectedObject(GameObject selectedSceneObject, Type valueType, Guid oldUniqueName)
+        {
+            Guid newGuid = Guid.Empty;
+
+            if (selectedSceneObject != null)
+            {
+                if (selectedSceneObject.GetComponent(valueType) != null)
+                {
+                    if (typeof(ISceneObject).IsAssignableFrom(valueType))
+                    {
+                        newGuid = GetUniqueIdFromSceneObject(selectedSceneObject);
+                    }
+                    else if (typeof(ISceneObjectProperty).IsAssignableFrom(valueType))
+                    {
+                        newGuid = GetUniqueIdFromProcessProperty(selectedSceneObject, valueType, oldUniqueName);
+                    }
+                }
+                else
                 {
-                    RevertableChangesHandler.CollapseUndoOperations(undoGroupName);
+                    // TODO handle non-PSO
```

Comment:
Do we need to do this bevore 4.0?

Reply:
I'd say so, and there is a ticket. However, VR Builder core-only doesn't use process variables, so this is used only by some add-ons.